docs/schema_plugins.md

```diff
@@ -1 +1,234 @@
+ 'version': nomad_simulations.general.Program.version:Quantity,
+ 'link': nomad_simulations.general.Program.link:Quantity,
+ 'version_internal': nomad_simulations.general.Program.version_internal:Quantity,
+ 'compilation_host': nomad_simulations.general.Program.compilation_host:Quantity}
+ ```
+
+ Indeed, this class contains `name` and `version` information, along with other quantities that make sense in the context of the H5MD-NOMAD creator. So, in this case we can simply reuse this class. We need to simply add a new section within the `Simulation()` class for the H5MD-NOMAD file creator that is of type `Program()`. In `schema_packages/<parser_name>_schema.py`:
+
+ ```python
+class Simulation(nomad_simulations.schema_packages.general.Simulation):
+
+    x_h5md_creator = SubSection(
+        sub_section=nomad_simulations.schema_packages.general.Program.m_def
+    )
+```
+
+Here, we create a new class called `Simulation()` that inherits from `Simulation()` from `nomad-simulations` while adding the additional `x_h5md_creator` section of type `Program()`. Note that the `x_<program_name>_<quantity_name>` notation is our standard notation for specifying program-specific quantities that are not necessarily generalizable.
+
+Now we can use this new class within our parser to store the creator information:
+
+```python
+from nomad_simulations.schema_packages.general import Program
+from nomad_simulations.schema_packages/<parser_name>_schema.py import Simulation
+
+simulation = Simulation()
+simulation.x_h5md_creator = Program(
+    name=group_h5md_dict.get('h5md_creator_name'),
+    version=group_h5md_dict.get('h5md_creator_version'),
+)
+```
+
+We now need to extend the schema further to store the h5md-nomad author information. For this, let's make a new class with `name` and `email` quantities in `schema_packages/<parser_name>_schema.py`:
+
+```python
+class Author(ArchiveSection):
+    """
+    Contains the specifications of the program.
+    """
+
+    name = Quantity(
+        type=str,
+        shape=[],
+        description="""
+        Specifies the name of the author who generated the h5md file.
+        """,
+    )
+
+    email = Quantity(
+        type=str,
+        shape=[],
+        description="""
+        Author's email.
+        """,
+    )
+```
+
+Again, we need to add this to our custom `Simulation()` class:
+
+ ```python
+class Simulation(nomad_simulations.schema_packages.general.Simulation):
+
+    x_h5md_creator = SubSection(
+        sub_section=nomad_simulations.schema_packages.general.Program.m_def
+    )
+
+    x_h5md_author = SubSection(sub_section=Author.m_def)
+```
+
+And, in our parser:
+
+```python
+from nomad_simulations.schema_packages/<parser_name>_schema.py import Simulation, Author
+
+simulation = Simulation()
+simulation.x_h5md_author = Author(
+    name=group_h5md_dict.get('h5md_author_name'),
+    email=group_h5md_dict.get('h5md_author_email'),
+)
+
+```
```

Comment:
Firstly, I don't see the added quality of this example: it's a repeat of the former. Maybe you could give it as an assignment?

Moreover, I think this particular field is kinda hairy: we have sections for listing authors, i.e. `metainfo`. I'd argue that one should extend that one (more involved), once Area D has the user semantics merged. I would choose a different field.